app/scripts/metamask-controller.js

```diff
@@ -5233,6 +5253,47 @@ export default class MetamaskController extends EventEmitter {
     }
   }
 
+  /**
+   * Discovers and creates accounts for the given keyring id.
+   *
+   * @param {string} id - The keyring id to discover and create accounts for.
+   * @returns {Promise<Record<string, number>>} Discovered account counts by chain.
+   */
+  async discoverAndCreateAccounts(id) {
+    try {
+      await this.userStorageController.setHasAccountSyncingSyncedAtLeastOnce(
+        false,
+      );
+      await this.userStorageController.setIsAccountSyncingReadyToBeDispatched(
+        false,
+      );
+
+      // If no keyring id is provided, we assume one keyring was added to the vault
+      const keyringIdToDiscover =
+        id || this.keyringController.state.keyrings[0].metadata.id;
+
+      const multichainAccountWallet = this.messenger.call(
+        'MultichainAccountService:getMultichainAccountWallet',
+        keyringIdToDiscover,
+      );
+
+      return await multichainAccountWallet.discoverAndCreateAccounts();
```

Comment:
This one will miss the `Bitcoin` key. Should we have an object result and use `Object.assign` here?

```suggestion
      const result =
        await multichainAccountWallet.discoverAndCreateAccounts();
      
      // Use default values, since we won't have a result for Bitcoin accounts yet.
      return Object.assign({
        Bitcoin: 0,
        Solana: 0,
      }, result};
```